Add static section whitelist info to api docs generation

jdconrad opened the pull request:
This change adds imported methods, class bindings, and instance bindings to the documentation generation for the Painless Context APIs.

jdconrad committed:
Add static section whitelist info to api docs generation

elasticmachine commented:
Pinging @elastic/es-core-infra

jdconrad committed:
Merge branch 'master' into staticdoc

jdconrad committed:
Merge branch 'master' into staticdoc

jdconrad commented:
@elasticmachine run elasticsearch-ci/1

jaymode committed:
Update TLS ciphers and protocols for JDK 11 (#41808)

* Update TLS ciphers and protocols for JDK 11 (#41385)

This commit updates the default ciphers and TLS protocols that are used
after the minimum supported JDK is JDK 11. The conditionals around
TLSv1.3 and 256-bit cipher support have been removed. JDK 11 no longer
requires an unlimited JCE policy file for 256 bit cipher support and
TLSv1.3 is supported in JDK 11+. New cipher support has been introduced
in the newer JDK versions as well. The ciphers are ordered with PFS
ciphers being most preferred, then AEAD ciphers, and finally those with
mainstream hardware support.

* Fixes for TLSv1.3 on JDK11

* fix for JDK-8212885

romseygeek committed:
Correct spelling of MockLogAppender.PatternSeenEventExpectation (#41893)

The class was called PatternSeenEventExcpectation. This commit
is a straight class rename to correct the spelling.

DaveCTurner committed:
Handle serialization exceptions during publication (#41781)

Today if an exception is thrown when serializing a cluster state during
publication then the master enters a poisoned state where it cannot publish any
more cluster states, but nor does it stand down as master, yielding repeated
exceptions of the following form:

```
failed to commit cluster state version [12345]
org.elasticsearch.cluster.coordination.FailedToCommitClusterStateException: publishing failed
        at org.elasticsearch.cluster.coordination.Coordinator.publish(Coordinator.java:1045) ~[elasticsearch-7.0.0.jar:7.0.0]
        at org.elasticsearch.cluster.service.MasterService.publish(MasterService.java:252) [elasticsearch-7.0.0.jar:7.0.0]
        at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:238) [elasticsearch-7.0.0.jar:7.0.0]
        at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:142) [elasticsearch-7.0.0.jar:7.0.0]
        at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [elasticsearch-7.0.0.jar:7.0.0]
        at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [elasticsearch-7.0.0.jar:7.0.0]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:681) [elasticsearch-7.0.0.jar:7.0.0]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252) [elasticsearch-7.0.0.jar:7.0.0]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215) [elasticsearch-7.0.0.jar:7.0.0]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_144]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_144]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_144]
Caused by: org.elasticsearch.cluster.coordination.CoordinationStateRejectedException: cannot start publishing next value before accepting previous one
        at org.elasticsearch.cluster.coordination.CoordinationState.handleClientValue(CoordinationState.java:280) ~[elasticsearch-7.0.0.jar:7.0.0]
        at org.elasticsearch.cluster.coordination.Coordinator.publish(Coordinator.java:1030) ~[elasticsearch-7.0.0.jar:7.0.0]
        ... 11 more
```

This is because it already created the publication request using
`CoordinationState#handleClientValue()` but then it fails before accepting it.
This commit addresses this by performing the serialization before calling
`handleClientValue()`.

Relates #41090, which was the source of such a serialization exception.

lcawl committed:
[DOCS] Fix callouts for dataframe APIs (#41904)

jdconrad committed:
Merge branch 'master' into staticdoc

jdconrad committed:
Add new api

matriv committed:
SQL: Remove CircuitBreaker from parser (#41835)

The CircuitBreaker was introduced as means of preventing a
`StackOverflowException` during the build of the AST by the parser.

The ANTLR4 grammar causes a weird behaviour for a Parser Listener.
The `enterEveryRule()` method is often called with a different parsing
context than the respective `exitEveryRule()`. This makes it difficult
to keep track of the tree's depth, and a custom Map was used as an
attempt of matching the contextes as they are encounter during `enter`
and during `exit` of the rules.

This approach had 2 important drawbacks:
1. It's hard to maintain this custom Map as the grammar changes.
2. The CircuitBreaker could often lead to false positives which caused
valid queries to return an Exception and prevent them from executing.

So, this removes completely the CircuitBreaker which is replaced be
a simple handling of the `StackOverflowException`

Fixes: #41471

jdconrad commented:
@elasticmachine run elasticsearch-ci/1

williamrandolph committed:
Allow unknown task time in QueueResizingEsTPE (#41810)

* Allow unknown task time in QueueResizingEsTPE

The afterExecute method previously asserted that a TimedRunnable task
must have a positive execution time. However, the code in TimedRunnable
returns a value of -1 when a task time is unknown. Here, we expand the
logic in the assertion to allow for that possibility, and we don't
update our task time average if the value is negative.

Fixes #41448

* Add a failure flag to TimedRunnable

In order to be sure that a task has an execution time of -1 because of
a failure, I'm adding a failure flag boolean to the TimedRunnable class.
If execution time is negative for some other reason, an assertion will
fail.

rjernst commented on modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextDocGenerator.java:
```diff
@@ -123,7 +129,39 @@ public static void main(String[] args) throws IOException {
+
+        for (PainlessContextInfo contextInfo : contextInfos) {
+            for (PainlessContextMethodInfo methodInfo : contextInfo.getImportedMethods()) {
+                staticInfoCounts.compute(methodInfo, (k, v) -> v == null ? 1 : v + 1);
```
I think this can be done more simply with `merge`:

```
staticInfoCounts.merge(methodInfo, 1, Integer::sum)
```

jdconrad replied:
Good catch.  Fixed.

rjernst reviewed (approved):
LGTM, one suggestion

rjernst committed:
Cleanup versioned deprecations in analysis (#41560)

This commit removes versioned logic in analyzer creation that is no
longer relevant for 8.0.

jdconrad committed:
Merge branch 'master' into staticdoc

jdconrad commented:
@rjernst Thank you for the review.  Will commit as soon as CI completes.

jdconrad committed:
Change compute to merge.

jasontedor committed:
Remove ml-cpp binary files

This commit removes some ml-cpp binary files that were commited to the
repository.

jdconrad committed:
Merge branch 'master' into staticdoc

jdconrad committed:
Merge branch 'master' into staticdoc

jdconrad commented:
@elasticmachine run elasticsearch-ci/2
